integrations/opencost/integration.py

```diff
@@ -92,6 +103,15 @@ class OpencostSelector(Selector):
     resolution: ResolutionField | None = Field(
         description="Duration to use as resolution in Prometheus queries",
     )
+    cloudcost_aggregate: CloudCostAggregateField | None = Field(
+        description="Field by which to aggregate the results of cloudcost",
+        alias="cloudcostAggregate",
+    )
```

Comment:
usually, we create different selector that will be applied only for certain kinds please take a look at how other integrations used literal strings to create custom validations (search for kind: Literal[....)

It will help you to remove the cloudcost naming prefix

Reply:
Literal strings would have worked, except that they can be separated by comma in any order. Only regex can handle that validation.

Reply:
The API docs: [https://www.opencost.io/docs/integrations/api#allocation](https://www.opencost.io/docs/integrations/api#allocation)

Reply:
that's not what im talking about
u added a filter that should only apply for cloud cost resources
therefore change the model to validate this field only for cloud cost resources